pkg/monitor/datapath_policy.go

```diff
@@ -61,6 +63,36 @@ type PolicyVerdictNotify struct {
 	// data
 }
 
+// DecodePolicyVerdictNotify will decode 'data' into the provided PolicyVerdictNotify structure
+func DecodePolicyVerdictNotify(data []byte, pvn *PolicyVerdictNotify) error {
+	return pvn.decodePolicyVerdictNotify(data)
+}
+
+func (n *PolicyVerdictNotify) decodePolicyVerdictNotify(data []byte) error {
+
+	if len(data) < PolicyVerdictNotifyLen {
+		return errors.New("unexpected length of data to decode into PolicyVerdictNotify")
+	}
+
+	n.Type = data[0]
+	n.SubType = data[1]
+	n.Source = byteorder.Native.Uint16(data[2:4])
+	n.Hash = byteorder.Native.Uint32(data[4:8])
+	n.OrigLen = byteorder.Native.Uint32(data[8:12])
+	n.CapLen = byteorder.Native.Uint16(data[12:14])
+	n.Version = byteorder.Native.Uint16(data[14:16])
+	n.RemoteLabel = identity.NumericIdentity(byteorder.Native.Uint32(data[16:20]))
+	n.Verdict = int32(byteorder.Native.Uint32(data[20:24]))
+	n.DstPort = byteorder.Native.Uint16(data[24:26])
+	n.Proto = data[26]
+	n.Flags = data[27]
+	n.AuthType = data[28]
+	n.Pad1 = data[29]
+	n.Pad2 = byteorder.Native.Uint16(data[30:32])
```

Comment:
I would be inclined to skip decoding padding here since they are never read, @gandro what do you think?

Reply:
I prefer to leave it in. I think we should keep the code as mechanical and stupid as possible, i.e. it shouldn't assume anything about the content and how it is used, and be as close to `binary.Read` as possible.

Plus having the decoding already in place will make it easier if we ever actually change the padding to contain a real value. 

But no strong opinion, i.e. I'm also fine if we remove it.